### PR TITLE
Retry failed registry HTTP requests

### PR DESCRIPTION
**Pull request by ivanilves**
_Sometimes HTTP requests to Docker registry fail (the more tags we query the more :boom: :scream_cat:).
However we could retry on errors, if that has at least some sense (not for the 4XX of course)!_

Solving this old and nasty https://github.com/ivanilves/lstags/issues/29

## NB!
Please see also: https://github.com/ivanilves/lstags/issues/85#issuecomment-342890182

# GIF

### CODE REVIEW

File: tag/remote/remote.go [hunk 1]
```diff
@@ -73,7 +75,7 @@ func httpRequest(url, authorization, mode string) (*http.Response, error) {
 		return nil, err
 	}
 	if resp.StatusCode != 200 {
-		return nil, errors.New("Bad response status: " + resp.Status + " >> " + url)
+		return resp, errors.New("Bad response status: " + resp.Status + " >> " + url)
```
Comment (ivanilves) [inline tag/remote/remote.go]: This was the culprit of a "panic" failure, we should return resp, not `nil`, if we want to communicate request status to `httpRetriableRequest` wrapper function!

File: tag/remote/remote.go [hunk 2]
```diff
@@ -23,13 +23,15 @@ import (
 // ConcurrentRequests defines maximum number of concurrent requests we could maintain against the registry
 var ConcurrentRequests = 32
 
+// RetryRequests is a number of retries we do in case of request failure
+var RetryRequests = 0
+
+// RetryDelay is a delay between retries of failed requests to the registry
+var RetryDelay = 5 * time.Second
```
Comment (zmalik) [inline tag/remote/remote.go]: why not a const?

File: tag/remote/remote.go [hunk 3]
```diff
@@ -91,6 +93,43 @@ func httpRequest(url, authorization, mode string) (*http.Response, error) {
...
+		if try < tries {
+			fmt.Printf(
+				"Will retry '%s' [%s] in a %v\n=> Error: %s\n",
+				url,
+				mode,
+				RetryDelay,
+				err.Error(),
+			)
+
+			time.Sleep(RetryDelay)
```
Comment (zmalik) [inline tag/remote/remote.go]: easy peasy backoff to save the internet?

File: tag/remote/remote.go (hunk 2 above)
Comment (ivanilves) [reply tag/remote/remote.go]: because we need to change it on runtime @zmalik  :wink:

**Pull request merged**